src/main/scala/cromwell/binding/types/WdlType.scala

```diff
@@ -54,6 +56,32 @@ trait WdlType {
 
     ast.wdlValue(this, wdlSyntaxErrorFormatter)
   }
+
+  def invalid(operation: String) = Failure(new WdlExpressionException(s"Cannot perform operation: $operation"))
+  def add(rhs: WdlType): Try[WdlType] = invalid(s"$this + $rhs")
+  def subtract(rhs: WdlType): Try[WdlType] = invalid(s"$this - $rhs")
+  def multiply(rhs: WdlType): Try[WdlType] = invalid(s"$this * $rhs")
+  def divide(rhs: WdlType): Try[WdlType] = invalid(s"$this / $rhs")
+  def mod(rhs: WdlType): Try[WdlType] = invalid(s"$this % $rhs")
+  def equals(rhs: WdlType): Try[WdlType] = invalid(s"$this == $rhs")
+  def notEquals(rhs: WdlType): Try[WdlType] = equals(rhs) map {x => WdlBooleanType}
+  def lessThan(rhs: WdlType): Try[WdlType] = invalid(s"$this < $rhs")
+  def lessThanOrEqual(rhs: WdlType): Try[WdlType] = (lessThan(rhs), equals(rhs)) match {
```

Comment:
I looked into it a bit and I don't think it's possible to define any of these at on the `WdlType` or `WdlPrimitiveType` classes because the specific implementations vary too much.  For example... `Float == Integer` is valid, but `Float == File` is not a valid operation... however `String == File` is.